libcontainer/rootfs_linux.go

```diff
@@ -77,6 +82,16 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig, mountFds mountFds) (
 			mountConfig.fd = nil
 		}
 
+		// We validated before we can access idmapFds[i].
+		mountConfig.idmapFd = nil
```

Comment:
No need to explicitly set it to `nil`, as this is the default value.

Reply:
@kolyshkin nope, we need to. The mount config struct is created outside the loop, so if we don't re-set it to nil on each operation, we can use an old fd.

This actually caused a bug that you fixed by setting it the mountConfig.fd to nil in the else case: https://github.com/opencontainers/runc/commit/d370e3c04660201e72ba6968342ce964c31a2d7f

The fix I proposed that you didn't like was moving the creation of the mountConfig struct inside the loop, so we don't need to care about re-setting it. But you didn't like that, so we need to assign nil when the mount doesn't have an fd.